mysql/awsmysql/awsmysql.go

```diff
@@ -129,16 +122,14 @@ func (c *connector) Connect(ctx context.Context) (driver.Conn, error) {
 			c.sem <- struct{}{} // release
 			return nil, fmt.Errorf("connect RDS: register TLS: %v", err)
 		}
-		cfg := &mysql.Config{
-			Net:                     "tcp",
-			Addr:                    c.addr,
-			User:                    c.user,
-			Passwd:                  c.password,
-			TLSConfig:               tlsConfigName,
-			AllowCleartextPasswords: true,
-			AllowNativePasswords:    true,
-			DBName:                  c.dbName,
+		cfg, err := mysql.ParseDSN(c.dsn)
```

Comment:
This requires parsing the same string on every new connection. Better to parse the DSN during `OpenMySQLURL` and then clone it during `Connect`.

Reply:
Agree. But we have to call [mysql Config.normalize](https://github.com/go-sql-driver/mysql/blob/6844171c7b652e0a8271a909fc48f6a32492d04f/dsn.go#L98) here to call [mysql.getTLSConfigClone](https://github.com/go-sql-driver/mysql/blob/d9aa6d3abe0f1072adb4aa71e1eb1d3eb3d3921c/utils.go#L82) to use SSL. 
So, I think it should be called here even if it is redundant.

Reply:
I don't understand this statement. What's the end effect of this unexported field? Does this mean the same `*mysql.Config` can't be used for multiple connections? That seems like a pretty unfortunate flaw in the underlying MySQL library and seems like it should be fixed upstream.

Reply:
@ymt2 Did you look at the https://github.com/google/go-cloud/blob/master/mysql/gcpmysql/gcpmysql.go#L121 where this has also been done.  I have used it in several projects and it is work well for me.